pages/[type]/[id]/versions.vue

```diff
@@ -161,15 +153,32 @@ const metaDescription = computed(
 
 const route = useRoute()
 const currentPage = ref(Number(route.query.p ?? 1))
-const filteredVersions = shallowRef(props.versions)
+const filteredVersions = computed(() => {
```

Comment:
only comment is this looks like a sizable chunk of duplicated code

Reply:
yeah I think we can fix this in the future, but it felt weird putting it in utils for such a specific feature